xbmc/network/EventServer.cpp

```diff
@@ -203,7 +203,10 @@ void CEventServer::Run()
   // add our socket to the 'select' listener
   listener.AddSocket(m_pSocket);
 
-  m_bRunning = true;
+  {
+    CSingleLock lock(m_critSection);
```

Comment:
The way it was used before (e.g. not caring about parallel codeblocks) indicates that an atomic bool might be enough. Locking the critSection might have side effects on the way the class is used.